files/js.js

```diff
@@ -0,0 +1,161 @@
+	    return false;
+	}
+	
+	if(user.profile.hasOwnProperty("email")) {
+	    score += stringSimilarity.compareTwoStrings(user.profile["email"].toLowerCase(), helloasso.last_name.toLowerCase());
+	    count +=1;
+	}
+	
+	if(user.profile.hasOwnProperty("first_name") && user.profile.first_name.length > 0 &&
+	   user.profile.hasOwnProperty("last_name") && user.profile.last_name.length > 0) {
+	    score += stringSimilarity.compareTwoStrings(user.profile["first_name"].toLowerCase(), helloasso.first_name.toLowerCase())
+	    score += stringSimilarity.compareTwoStrings(user.profile["last_name"].toLowerCase(), helloasso.last_name.toLowerCase());
+	    count +=2;
+	} else if(user.profile.hasOwnProperty("real_name") && user.profile.real_name.length > 0) {
+	    score += stringSimilarity.compareTwoStrings(user.profile.real_name.toLowerCase(),
+							helloasso.first_name.toLowerCase() + " " + helloasso.last_name.toLowerCase())
+	    count +=1;
+	} else if(user.profile.hasOwnProperty("display_name") && user.profile.display_name.length > 0) {
+	    score += stringSimilarity.compareTwoStrings(user.profile["display_name"].toLowerCase(),
+							helloasso.first_name.toLowerCase() + " " + helloasso.last_name.toLowerCase())
+	    count +=1;
+	} else if(user.hasOwnProperty("real_name") && user["real_name"].length > 0) {
+	    score += stringSimilarity.compareTwoStrings(user["real_name"].toLowerCase(),
+							helloasso.first_name.toLowerCase() + " " + helloasso.last_name.toLowerCase())
+	    count +=1;
+	} else if(user.hasOwnProperty("display_name") && user["display_name"].length > 0) {
+	    score += stringSimilarity.compareTwoStrings(user["display_name"].toLowerCase(),
+							helloasso.first_name.toLowerCase() + " " + helloasso.last_name.toLowerCase())
+	    count +=1;
+	} else {
+	    console.log("User profile is not complete.");
+	    console.log(user);
+	    return false;
+	}
+
+	user.score = score/(1.*count);
+	return true;
+    });
+    
+    scores.sort(function(a, b) {
+	return a.score < b.score;
+    });
+    
+    scores.slice(0, 3).forEach((user) => {
+	var slack = $(`<div slackID="${user.id}" class='col-3 border-bottom border-2 border-dark'></div>`).appendTo(row);
+	var color = `rgba(0,255,0,${Math.pow(user.score,2)})`;
+	slack.css('background-color', color);
+	
+	if(user.deleted) {
+	    slack.css("background", `repeating-linear-gradient(
+  -45deg,
+  #ffffff,
+  #ffffff 10px,
+  ${color} 10px,
+  ${color} 20px
+)`);
+	}
+
+	
+	$(slack).append(
+	    user["id"] + "</br>" + 
+		userprofileName(user) + "</br>" + 
+		user.profile["email"]
+	);
+	
+	slack.click(function() {
+	    var slackID = $(this).attr("slackID");
+	    var helloassoID = $($($(this).parent()).children()[0]).attr("helloassoID");
+	    
+	    if(!$(this).hasClass("matched")) {
+		$(this).css('background-color', "rgba(0,0,255,0.4)");
+		$(this).addClass("matched");
+		$($(this).parent()).children("[slackID]").not(".matched").remove();
+		$(`[slackID=${slackID}]`).not(this).hide();
```

Comment:
what does this do? I don't understand what we are selecting and I hence don't understand what we are hiding...

Reply:
We are removing all the "green boxes" related to the same slack account but `this`. 
(As a slack account may be related to only one helloasso registration.)